keps/sig-instrumentation/2845-deprecate-klog-specific-flags-in-k8s-components/README.md

```diff
@@ -333,3 +333,202 @@ features makes their future removal much harder.
+restore previous behavior. 3 releases should be enough heads up for users to
+make necessary changes to avoid breakage. 
+
+In case of changing log output, we will introduce a temporary flag 
+`--logtostdout` that will allow users to flip back to previous behavior. 
+This flag too will be marked as deprecated and removed with same period.
+
+### Version Skew Strategy
+
+Proposed changes have no impact on cluster that would require coordination. 
+They only affect binary configuration and logs are written, which don't impact
+other components in cluster. Users might be required to change flags passed to 
+k8s binaries, but this can be done one by one independently of other components.
+
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+For stdout/stderr log split feature we are introducing dedicated flag 
+`--logtostdout`, which can be used to enable and rollback the feature.
+
+As this feature will be releases in stages, let's discuss them separately:
+* Alpha - feature is available, users can enable the feature by passing 
+  `--logtostdout` flag to binaries.
+* Beta - feature is enabled by default. Users can rollback by passing 
+  `--logtostdout=false` flag to binaries. Old configuration is deprecated.
+* GA - Old configuration is no longer supported. This is ok wait 3 releases to
+  uphold Kubernetes deprecation policy.
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+- [ ] Feature gate (also fill in values in `kep.yaml`)
+  - Feature gate name:
+  - Components depending on the feature gate:
+- [x] Other
+  - Describe the mechanism: Passing command line flag to K8s component binaries.
+  - Will enabling / disabling the feature require downtime of the control
+    plane?
+    **Yes, for apiserver it will require a restart, which can be considered a 
+    control plane downtime in non highly available clusters**
+  - Will enabling / disabling the feature require downtime or reprovisioning
+    of a node? (Do not assume `Dynamic Kubelet Config` feature is enabled).
+    **Yes, it will require restart of Kubelet**
+
+###### Does enabling the feature change any default behavior?
+
+Yes, this feature will change what flags are available in K8s binaries and how
+logs are written. For flags change will go through K8s flag deprecation policy,
+For logs we will introduce a flag to allow users to rollback the change.
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+Yes, there is no impact on cluster state. Logs should also be unaffected as logs
+will be only written to stderr.
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+No impact on cluster. 
+
+###### Are there any tests for feature enablement/disablement?
+
+New logging configuration will be tested as it will become a default in E2e 
+tests. Testing disabled feature will be handled in klog unit tests.
```

Comment:
Have you reviewed test-infra manifests to ensure this change won't break e2e tests in prow? I know right now the node ones do some trickery with systemd to redirect stderr but not stdout.

Reply:
Can you provide what systemd trickery you mean?

Reply:
https://github.com/kubernetes/kubernetes/pull/104047/files#r681146455